bigquery_etl/routine/parse_routine.py

```diff
@@ -191,7 +191,10 @@ def from_file(cls, path):
         routines = get_routines(project)
         dependencies = []
         for udf in routines:
-            if udf["name"] in "\n".join(definitions):
+            persistent_name = ".".join([f"`{name}`" for name in udf["name"].split(".")])
+            if udf["name"] in "\n".join(definitions) or persistent_name in "\n".join(
+                definitions
+            ):
```

Comment:
Couldn't the UDF identifiers be quoted multiple different ways? (`` `foo`.`bar` ``, `` `foo.bar` ``, `` `foo`.bar ``)

Here's a possible regular expression approach that handles the quoting variations, as well as guards against accidentally matching subsections of UDF identifiers (the existing logic would have a `bar.baz` UDF match a `foobar.baz` reference, or a `foo.bar` UDF match a `foo.barbaz` reference).
```suggestion
            udf_re = re.compile(
                r"\b"
                + r"\.".join(f"`?{name}`?" for name in udf["name"].split("."))
                + r"\("
            )
            if udf_re.search("\n".join(definitions)):
```

Reply:
yup, a regex seems like the best option here